client/js/color.js

```diff
@@ -0,0 +1,101 @@
+export function toHex(inputColor) {
+  const hexPattern = /^#([A-Fa-f0-9]{6})$/;
+  if (hexPattern.test(inputColor)) {    
+    return inputColor;
+  }
+  const canvas = document.createElement('canvas');
+  const ctx = canvas.getContext('2d');
+  ctx.fillStyle = inputColor;
+  const hexColor = ctx.fillStyle;  
```

Comment:
There is a chance that the fillstyle will not be a `#rrggbb` color. If an invalid value is provided the default of `#000000` is returned, but gradients and patterns are also valid fill styles and CSS 4 colors include LCH and LAB color functions which can result in colors outside sRGB. Apple has already created devices that support colors outside sRGB and added these color spaces to Safari.